concentration/Scene/SpinningBoxAndBalls.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  SpinningBoxAndBalls.swift
+//  concentration
+//
+//  Created by Olha Dzhyhirei on 1/30/23.
+//
+
+import SpriteKit
+
+class SpinningBoxAndBalls: SKScene {
+    
+    var BallQuantity = 16
+    
+    enum GameState { case startGame, playing }
+    var gameState = GameState.startGame
+    var box: SKShapeNode!
+    
+    override func didMove(to view: SKView) {
+        self.anchorPoint = .init(x: 0, y: 0)
+        self.scene?.backgroundColor = .link
+        setupStartLabel()
+    }
+    
+    // MARK:  game state: startGame
+    
+    func setupStartLabel() {
+        let startLabel = SKLabelNode()
+        startLabel.name = "start label"
+        startLabel.fontColor = .black
+        startLabel.fontSize = 26
+        startLabel.text = "Tap anywhere to start"
+        startLabel.fontName = "AvenirNext-Bold"
+        startLabel.zPosition = 2
+        startLabel.position = .init(x: self.frame.midX, y: self.frame.midY)
+        let action = SKAction.sequence([.scale(by: 1.25, duration: 1), .scale(by: 0.8, duration: 1)])
+        startLabel.run(.repeatForever(action))
+        self.addChild(startLabel)
+        
+    }
+    
+    // MARK: game state: setupPlaying
+    
+    func createBall(ofRadius radius: CGFloat) -> SKSpriteNode {
+        let ball = SKSpriteNode(texture: .init(imageNamed: "ball_04"))
+        ball.size = .init(width: radius * 2, height: radius * 2)
+        ball.physicsBody = .init(circleOfRadius: radius)
+        ball.physicsBody?.isDynamic = false
+        ball.physicsBody?.restitution = 1
+        ball.physicsBody?.friction = 0.05
+        ball.name = "ball"
+        return ball
+    }
+    
+    func createteBox() -> SKShapeNode {
+        let side = frame.width < frame.height ? frame.width / sqrt(2) : frame.height / sqrt(2)
+        let box = SKShapeNode(rectOf: .init(width: side, height: side))
+        box.strokeColor = SKColor.red
+        box.physicsBody = .init(edgeLoopFrom: box.frame.inset(by: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)))
+        box.position = .init(x: frame.midX, y: frame.midY)
+        self.addChild(box)
+        return box
```

Comment:
Maybe add some color to box outside, so its different inside and outside
Plus consider it rotated for 90 from the beging so balls fall to the edge

Reply:
done